app.rb

```diff
@@ -0,0 +1,109 @@
+require_relative './book'
+require_relative './student'
+require_relative './teacher'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def all_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author} "
+    end
+    nil
+  end
+
+  def all_people
+    @people.each do |person|
+      puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}"
+    end
+    nil
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)?: '
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    student = Student.new age, name, parent_permission: get_permission(permission)
+    @people.push(student)
+    nil
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
```

Comment:
- [ ] Good job but this method is not following single responsibility principle. because all requests for input and input validation should've been placed in the methods whose responsibilities are to request or validate input and then you can call that method here in create_person if you wish, so kindly consider adding a method just for this purpose. you can read [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

Reply:
Thanks for your feedback, we have created a method that is called request_option and now we are calling that method here instead of requesting the option directly in create_teacher 🚀.